src/leaflet_d3.js

```diff
@@ -242,13 +242,11 @@ L.CartoDBd3Layer = L.TileLayer.extend({
     this.tileLoader._tilesLoading = {}
     for (i = 0, len = tiles.length; i < len; i++) {
       tile = tiles[i]
-
-      if (!tile.complete) {
-        tile.onload = L.Util.falseFn
-        tile.onerror = L.Util.falseFn
-        tile.src = L.Util.emptyImageUrl
-      }
       tile.parentNode.removeChild(tile)
     }
+    for (var tileKey in this.provider.requests){
```

Comment:
I don't have much context, but I see a lot of `this.provider.something` here. That means this guy have a lot of details about how the provider is implemented internally. Instead, the provider could expose a method like `this.provider.abortPendingRequests()` (there might be a better name that would work for other type of providers).